lib/logthrsource/logthrsourcedrv.h

```diff
@@ -74,6 +71,9 @@ struct _LogThreadedSourceDriver
   LogThreadedSourceWorker *worker;
 
   const gchar *(*format_stats_instance)(LogThreadedSourceDriver *self);
+  LogThreadedSourceWorkerRunFunc run;
+  LogThreadedSourceWorkerRequestExitFunc request_exit;
+  LogThreadedSourceWorkerWakeupFunc wakeup;
```

Comment:
I think these belong to the worker.

I'm okay with keeping them here for this PR, but I'd like to sort these out in a follow-up PR, like this:
https://github.com/MrAnno/syslog-ng/commits/thrsource-worker

Reply:
I agree, however in that case the "self" method should point to a worker instance which it didn't in this case. And it would have opened the scope of these patches a lot.

In a way these are similar to the compat methods in LogThreadedDestDriver which also reside in the driver class and take a pointer to the driver as self.

Reply:
@bazsi mentioned that it might be worth keeping this interface for "simplified" threaded sources even after we allow multiple workers in threaded sources. I'm marking this comment as resolved, as I'll try refactoring/implementing the remaining bits in a separate PR.